diff --git src/nwdft/libxc/GNUmakefile src/nwdft/libxc/GNUmakefile
index b40903ee3a..7408c3b118 100644
--- src/nwdft/libxc/GNUmakefile
+++ src/nwdft/libxc/GNUmakefile
@@ -21,6 +21,9 @@ LIB_DEFINES += -DXC_MAJOR_VERSION=$(shell grep XC_MAJOR_VERSION $(LIBXC_INCLUDE)
 LIB_DEFINES += -DLIBXC_NP_SIZE=$(shell ./libxc_findsizenp.sh $(LIBXC_INCLUDE))
 LIB_INCLUDES += -I$(LIBXC_INCLUDE)
 endif
+ifdef LIBXC_MODDIR
+LIB_INCLUDES += -I$(LIBXC_MODDIR)
+endif
 LIB_INCLUDES += -I.
 
 
