--- src/tools/ga-5.7.2/tcgmsg/tcgmsg-mpi/nxtval-armci.c
+++ src/tools/ga-5.7.2/tcgmsg/tcgmsg-mpi/nxtval-armci.c
@@ -16,8 +16,8 @@
 static long pnxtval_counter_val;
 static long *pnxtval_counter=&pnxtval_counter_val;
 int nxtval_installed=0;
-extern int     *tcgi_argc;
-extern char  ***tcgi_argv;
+extern int     tcgi_argc;
+extern char  **tcgi_argv;
 #define INCR 1   /**< increment for NXTVAL */
 #define BUSY -1L /**< indicates somebody else updating counter*/
 #define NXTV_SERVER ((int)NNODES_() -1)
@@ -38,7 +38,7 @@ long NXTVAL_(long *mproc)
     int rc;
     int server = NXTV_SERVER;         /* id of server process */
 
-    install_nxtval(tcgi_argc, tcgi_argv);
+    install_nxtval(&tcgi_argc, &tcgi_argv);
 
     if (SR_parallel) {
         if (DEBUG_) {
