--- src/mcscf/mcscf_aux.F.orig	2020-03-06 10:48:12.850070560 -0800
+++ src/mcscf/mcscf_aux.F	2020-03-06 10:19:42.221115159 -0800
@@ -1196,7 +1196,11 @@
      $                         g_tmp, 1, nclosed, 1, nclosed )
       call ga_symmetrize( g_tmp )
 #ifdef PARALLEL_DIAG
+#ifdef SCALAPACK
+         call ga_pdsyev  (g_tmp, g_u, eps, 0)
+#else
       call ga_diag_std( g_tmp, g_u, eps )
+#endif
 #else
       call ga_diag_std_seq( g_tmp, g_u, eps )
 #endif
@@ -1221,7 +1225,11 @@
      $                         g_tmp, 1, nact, 1, nact )
       call ga_symmetrize( g_tmp )
 #ifdef PARALLEL_DIAG
+#ifdef SCALAPACK
+         call ga_pdsyev  (g_tmp, g_u, eps(aoff), 0)
+#else
       call ga_diag_std( g_tmp, g_u, eps(aoff) )
+#endif
 #else
       call ga_diag_std_seq( g_tmp, g_u, eps(aoff) )
 #endif
@@ -1247,7 +1255,11 @@
      $                          g_tmp, 1, nvir, 1, nvir )
       call ga_symmetrize( g_tmp )
 #ifdef PARALLEL_DIAG
+#ifdef SCALAPACK
+         call ga_pdsyev  (g_tmp, g_u, eps(aoff), 0)
+#else
       call ga_diag_std( g_tmp, g_u, eps(aoff) )
+#endif
 #else
       call ga_diag_std_seq( g_tmp, g_u, eps(aoff) )
 #endif
