--- src/config/makefile.h.orig
+++ src/config/makefile.h
@@ -1794,7 +1794,12 @@ endif
           DEFINES  += -DGCC46
         endif
         ifeq ($(GNU_GE_4_8),true)
-          FDEBUG =-O2 -g -fno-aggressive-loop-optimizations
+          ifeq ($(_CPU),ppc64le)
+          FDEBUG =-O0 -g 
+          else
+          FDEBUG =-O2 -g 
+          endif
+          FDEBUG +=-fno-aggressive-loop-optimizations
           FOPTIMIZE +=-fno-aggressive-loop-optimizations
           FFLAGS_FORGA += -fno-aggressive-loop-optimizations
           FOPTIONS += -Warray-bounds
--- src/solvation/GNUmakefile.orig
+++ src/solvation/GNUmakefile
@@ -6,14 +6,14 @@
          mnsol.o\
          solv_data.o\
          cosmo_smd_output.o\
-         cosmo_vem_output.o
+         cosmo_vem_output.o hnd_cosmo_lib.o
 
    OBJ_OPTIMIZE = cosmo_input.o \
          cosmo_initialize.o\
          cosmo_tidy.o\
          cosmo_def_radii.o\
          cosmo_charges.o\
-         hnd_cosmo_lib.o  grad_hnd_cos.o
+         grad_hnd_cos.o
 
    LIBRARY = libsolvation.a
 
--- src/nwdft/lr_tddft/GNUmakefile.orig
+++ src/nwdft/lr_tddft/GNUmakefile
@@ -1,6 +1,6 @@
 #$Id$
 
-OBJ = tddft_energy.o tddft_input.o tddft_init.o tddft_memory.o tddft_iguess.o \
+OBJ_OPTIMIZE = tddft_energy.o tddft_input.o tddft_init.o tddft_memory.o tddft_iguess.o \
 tddft_dump.o tddft_davidson.o tddft_transfm.o tddft_contract.o tddft_epsilon.o \
 tddft_subspace.o tddft_diagon.o tddft_residual.o tddft_analysis.o tddft_dftinit.o \
 tddft_finalize.o tddft_schmidt.o tddft_fold.o \
@@ -8,7 +8,7 @@ tddft_bdtensor.o tddft_patch.o tddft_civecs_phase.o tddft_oscstr.o \
 tddft_checkpoint.o tddft_restart.o tddft_s2.o tddft_simpleso.o \
 tddft_lanczos.o tddft_lanczos_tda.o tddft_dipole.o tddft_prod_v.o
 
-OBJ_OPTIMIZE = 
+#OBJ_OPTIMIZE = 
 
 LIBRARY = libnwdft.a
 

